### PR TITLE
Add document with list of features provided

### PR DESCRIPTION
**Pull request by davidor**
@unleashed feel free to add commits with your edits.

### CODE REVIEW

File: docs/features.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+- [Usage metrics of the Porta master account](#usage-metrics-of-the-porta-master-account)
+- [Internal management API](#internal-management-api)
+
+## Authorization of applications
+
+Requests are authorized based on some credentials that identify a service
+(provider key, service token) and some credentials that identify an application
+(app ID + app key, user key, oauth token).
+
+## Rate-limit of applications
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: Comment that it's not just a global rate-limit, but per metric... and multiple may apply.

Calculation of remaining quota per metric also (which is complex and led to some troubles also?)

File: docs/features.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Rate limits are applied per application. The limits can be per minute, hour,
+day, week, month, year, and "eternity". The limits are applied in calendar
+periods of time. So for example, if the limit is per year and the year is 2020,
+when applying the limit, Apisonator will only take into account the counters
+starting from Jan 1st 2020, not from a year ago.
+
+## Aggregation of usage stats
+
+Apisonator aggregates usage stats by the same periods used for rate-limiting
+(except for minute). It aggregates data by service and also by application. This
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: and by metric

File: docs/features.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+allows external system to query the usage for a given month, year, etc.
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: well - they can query the redis DB directly :-)

File: docs/features.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+## Aggregation of response codes
+
+Apisonator also aggregates response codes using the same periods used to
+aggregate usage stats. These response codes are the ones returned by the
+upstream API and are sent to Apisonator via APIcast or other integration
+mechanism (Istio adapter, Java plugin, etc.).
+
+## Alerts
+
+Apisonator sends usage alerts to Porta via a webhook. This alerts can be
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: These

File: docs/features.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+configured to be triggered when an application reaches 50% of it usage quota,
+80%, 90%, etc.
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: Fixed thresholds - or he thresholds are configurable?
How many? Unlimited number of thresholds.

Thresholds on any metric or just hits?

No thresholds on response codes...

File: docs/features.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+## Events
+
+Apisonator sends some events to Porta via a webhook. These events trigger when
+an application has traffic for the first time, and also, when it has traffic for
+the first time in the current day.
+
+## Real-time traffic
+
+Apisonator keeps track of the latest transactions of a service. This can be used
+to build a real-time traffic dashboard.
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: Fetched in chunks via the Internal API?

File: docs/features.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+When Apisonator receives invalid application keys, invalid metrics, or there is
+any other kind of error that suggests that the user did not configure the
+integration of his API with 3scale correctly, Apisonator stores these errors so
+they can be retrieved efficiently by Porta in the view that shows "integration
+errors".
+
+## Usage metrics of the Porta master account
+
+Apisonator keeps track of the authorize and reports call done for each service.
+In Porta, this is used to know the usage that each provider makes of the
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: So, strictly speaking not limited to Master only?

File: docs/features.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
+# Features
```
Comment (andrewdavidmackenzie) [inline docs/features.md]: Maybe some others - but either deprecated in SaaS, trying to deprecate or have never shipped or been supported on-prem (tokens, etc)

File: docs/features.md (hunk 6 above)
Comment (davidor) [reply docs/features.md]: Apisonator keeps a redis list with the latest 50 transactions, this allows us to fetch the latest transactions with just one redis call. There's an endpoint in the internal API to get them.

**Comment by davidor** [comment]
Thanks for the review :+1:
I have addressed your comments.

File: docs/features.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+Apisonator.
+
+- [Authorization of applications](#authorization-of-applications)
+- [Rate-limit of applications](#rate-limit-of-applications)
+- [Aggregation of usage stats](#aggregation-of-usage-stats)
+- [Aggregation of response codes](#aggregation-of-response-codes)
+- [Quotas](#quotas)
+- [Alerts](#alerts)
+- [Events](#events)
+- [Real-time traffic](#real-time-traffic)
```
Comment (unleashed) [inline docs/features.md]: TODO: remove this section.

Comment (davidor) [reply docs/features.md]: Done

**Pull request merged**